### PR TITLE
Added tests for file-upload-assessment-dashboard component

### PR DESCRIPTION
**Pull request by nicolasruscher**
### Checklist

- [X] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)

### Test Coverage
Test Coverage is only around 62% as most methods are private. Will be enhanced by implicit testing in future followup

### CODE REVIEW

File: src/test/javascript/spec/component/file-upload-assessment/file-upload-assessment-dashboard.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                JhiLanguageHelper,
+                { provide: ActivatedRoute, useValue: testRoute },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+            ],
+        })
+            .overrideTemplate(FileUploadAssessmentDashboardComponent, '')
+            .compileComponents()
+
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/file-upload-assessment/file-upload-assessment-dashboard.component.spec.ts:45]: The new line here looks a bit weird to me, but I don't know what the style policy for this is.

**Review by FrankeLukas — COMMENTED** [review]
The coverage report says 100% for the .html but only 62% for the .component.ts. Am I looking at the wrong report/file here?
grafik

**Review by ivanchimeno — APPROVED** [review]
The tests good!

File: src/test/javascript/spec/component/file-upload-assessment/file-upload-assessment-dashboard.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            // call
+            component.updateFilteredSubmissions([fileUploadSubmission]);
+
+            // check
+            expect(component.filteredSubmissions).toEqual([fileUploadSubmission]);
+        });
+    });
+
+    describe('cancelAssessment', () => {
+        it('test cancelAssessment', fakeAsync(() => {
```
Comment (stefanwaldhauser) [inline src/test/javascript/spec/component/file-upload-assessment/file-upload-assessment-dashboard.component.spec.ts:90]: fakeAsync and tick not needed.

File: src/test/javascript/spec/component/file-upload-assessment/file-upload-assessment-dashboard.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            // setup
+            const fileUpServicespy = spyOn(fileUploadSubmissionService, 'getFileUploadSubmissionsForExercise').and.returnValue({ body: { submission: [fileUploadSubmission] } });
+            const getSubSpy = spyOn<any>(component, 'getSubmissions').and.callThrough();
+
+            // call
+            component.ngOnInit();
+            tick();
+
+            // check
+            expect(component.submissions).toEqual([]);
```
Comment (stefanwaldhauser) [inline src/test/javascript/spec/component/file-upload-assessment/file-upload-assessment-dashboard.component.spec.ts:70]: I do not realy understand what you are testing here? This is the method you want to test:

image

You should test that exercise is set imo.

**Review by stefanwaldhauser — CHANGES_REQUESTED** [review]

**Comment by krusche** [comment]
closing for now, please reopen it when it is ready to merge

**Pull request closed without merging**